app/src/main/java/com/github/se/eduverse/viewmodel/ProfileViewModel.kt

```diff
@@ -84,6 +97,87 @@ open class ProfileViewModel(private val repository: ProfileRepository) : ViewMod
       }
     }
   }
+
+  open fun searchProfiles(query: String) {
+    // Cancel previous search if any
+    searchJob?.cancel()
+
+    if (query.isBlank()) {
+      _searchState.value = SearchProfileState.Idle
+      return
+    }
+
+    searchJob =
+        viewModelScope.launch {
+          _searchState.value = SearchProfileState.Loading
+          try {
+            // Add delay to avoid too many requests while typing
+            delay(300)
+            val results = repository.searchProfiles(query)
+            _searchState.value = SearchProfileState.Success(results)
+          } catch (e: Exception) {
+            _searchState.value = SearchProfileState.Error(e.message ?: "Search failed")
+          }
+        }
```

Comment:
Good method, canceling previous jobs is very clever as it avoids useless computation for searching previous text input.